pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -81,13 +81,13 @@ type InfinispanLoggingSpec struct {
 
 // InfinispanSpec defines the desired state of Infinispan
 type InfinispanSpec struct {
-	Replicas  int32                   `json:"replicas"`
-	Image     string                  `json:"image"`
-	Security  InfinispanSecurity      `json:"security"`
-	Container InfinispanContainerSpec `json:"container"`
-	Service   InfinispanServiceSpec   `json:"service"`
-	Logging   InfinispanLoggingSpec   `json:"logging"`
-	Expose    v1.ServiceSpec          `json:"expose"`
+	Replicas  int32                    `json:"replicas"`
+	Image     string                   `json:"image,optional,omitempty"`
+	Security  InfinispanSecurity       `json:"security,optional,omitempty"`
+	Container *InfinispanContainerSpec `json:"container,optional,omitempty"`
+	Service   InfinispanServiceSpec    `json:"service,optional,omitempty"`
```

Comment:
I'm not sure making this `omitempty` makes sense. Isn't this field always set either  to `cache` (by default) or `datagrid`?